test/stage2_fwd.jl

```diff
@@ -6,36 +6,34 @@ module stage2_fwd
         @test sin′(1.0) == cos(1.0)
     end
     let sin′′ = Diffractor.dontuse_nth_order_forward_stage2(Tuple{typeof(mysin), Float64}, 2)
-        # This broke some time between 1.10 and 1.11-DEV.10001
-        @test isa(sin′′, Core.OpaqueClosure{Tuple{Float64}, Float64}) broken=VERSION>=v"1.11-"
```

Comment:
These test cases about inferred OC types appear to work fine in v1.12, but they're still broken on v1.11. We might need to consider backporting some changes, or tweak the `broken` condition here.